src/data/debts.tsx

```diff
@@ -0,0 +1,178 @@
+import { atom, useRecoilState } from "recoil";
+import { CategoryI, port } from "./categories";
+import { userState } from "./authentication";
+import axios, { AxiosError } from "axios";
+import { createToastify } from "../helpers/toastify";
+import { useCookies } from "react-cookie";
+
+export type DebtType = 'debt' | 'loan'
+
+export interface DebtI {
+  _id?: string;
+  userId?: string;
+  type: DebtType;
+  beneficiary: string;
+  concept: string;
+  category: CategoryI;
+  amount: number;
+}
+
+export interface DebtFormI extends Omit<DebtI, "amount"> {
+  amount: string;
+}
+
+export const newDebtState = atom<DebtFormI | null>({
+  key: "newDebtState",
+  default: null,
+});
+
+export const debtsListState = atom<DebtI[]>({
+  key: "debtsListState",
+  default: [],
+});
+
+export const selectedDebtState = atom<DebtFormI | null>({
+  key: "selectedDebtState",
+  default: null,
+});
+
+export const useDebts = () => {
+  const [debtsList, setDebtsList] = useRecoilState(debtsListState);
+  const [user] = useRecoilState(userState);
+  const [cookies] = useCookies(["jwt"]);
+
+  const getDebts = async () => {
+    try {
+      if (user) {
+        const response = await axios.get(`${port}/debts/${user?._id}`, {
+          headers: {
+            Authorization: "Bearer " + cookies.jwt,
+          },
+        });
+        if (response.status === 200) {
+          setDebtsList(response.data);
+        } else {
+          createToastify({ text: "Debt or Loan not found", type: "error" });
+        }
+      }
+    } catch (err: unknown) {
+      if (err instanceof AxiosError) {
+        createToastify({
+          text: err.response?.data.message || err.message,
+          type: "error",
+        });
+        throw new Error(err.message);
+      }
+    }
+  };
+
+  const addDebt = async (newItem: DebtFormI) => {
+    try {
+      if (user) {
+        const parsedItem = {
+          ...newItem,
+          amount: parseInt(newItem.amount),
+          userId: user._id,
+        };
+        const response = await axios.post(`${port}/debts/`, parsedItem, {
+          headers: {
+            Authorization: "Bearer " + cookies.jwt,
+          },
+        });
+        if (response.status === 200) {
+          setDebtsList([...debtsList, response.data]);
+        } else {
+          createToastify({
+            text: "Add Debt not successful",
+            type: "error",
+          });
+        }
+      }
+    } catch (err: unknown) {
+      if (err instanceof AxiosError) {
+        createToastify({
+          text: err.response?.data.message || err.message,
+          type: "error",
+        });
+        throw new Error(err.message);
+      }
+    }
+  };
+
+  const editDebt = async (id: string, updatedItem: DebtFormI) => {
+    try {
+      if (user) {
```

Comment:
If there is not user. How are we handling that? Should be good to handle that error, in the case somethign strange happen with that